pgtest.go

```diff
@@ -56,6 +56,45 @@ func StartPersistent(folder string) (*PG, error) {
 	return start(New().DataDir(folder).Persistent())
 }
 
+// When the password is provided this function write the password
+// to a temp file, and returns its path.
+func createTempPasswordFile(password string, pgUID, pgGID int) (string, error) {
+	// Create a temporary file
+	tempFile, err := os.CreateTemp("", "pg_pwd_*")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temp file: %w", err)
+	}
+	tempFilePath := tempFile.Name()
+
+	// Set file permissions to 0600 (read/write for owner only)
+	if err := tempFile.Chmod(0600); err != nil {
+		tempFile.Close()
+		os.Remove(tempFilePath)
+		return "", fmt.Errorf("failed to set file permissions: %w", err)
+	}
+
+	// Write the password to the file
+	if _, err := tempFile.WriteString(password); err != nil {
+		tempFile.Close()
+		os.Remove(tempFilePath)
+		return "", fmt.Errorf("failed to write password to file: %w", err)
+	}
+
+	// Close the file
+	if err := tempFile.Close(); err != nil {
+		os.Remove(tempFilePath)
+		return "", fmt.Errorf("failed to close temp file: %w", err)
+	}
+
+	// Change the owner
+	err = os.Chown(tempFilePath, pgUID, pgGID)
+	if err != nil {
+		return "", fmt.Errorf("failed to set permissions on the temp file: %w", err)
+	}
+
+	return tempFilePath, nil
+}
```

Comment:
When specifying a password for the database, we need to write the password in a file and pass the path of the file to `initdb` using the flag `--pwdfile=<path>` along with the option `-A scram-sha-256`.

[read more about -A option](https://www.postgresql.org/docs/current/app-initdb.html#APP-INITDB-OPTION-AUTH)
[read more about --pwdfile flag](https://www.postgresql.org/docs/current/app-initdb.html#APP-INITDB-OPTION-PWFILE)